tests/your_first_contract.rs

```diff
@@ -0,0 +1,134 @@
+
+    let path = PathBuf::from("tests/your_first_contract.json");
+    let contract_class = ContractClass::try_from(path).unwrap();
+    let entry_points_by_type = contract_class.entry_points_by_type().clone();
+
+    let increase_balance_selector = entry_points_by_type
+        .get(&EntryPointType::External)
+        .unwrap()
+        .get(0)
+        .unwrap()
+        .selector()
+        .clone();
+
+    //* --------------------------------------------
+    //*    Create state reader with class hash data
+    //* --------------------------------------------
+
+    let ffc = DictStorage::new();
+    let contract_class_storage = DictStorage::new();
+    let mut contract_class_cache = HashMap::new();
+
+    //  ------------ contract data --------------------
+
+    let address = Address(1111.into());
+    let class_hash = [1; 32];
+    let contract_state = ContractState::new(class_hash, 3.into(), HashMap::new());
+
+    contract_class_cache.insert(class_hash, contract_class);
+    let mut state_reader = InMemoryStateReader::new(ffc, contract_class_storage);
+    state_reader
+        .contract_states_mut()
+        .insert(address.clone(), contract_state);
+
+    //* ---------------------------------------
+    //*    Create state with previous data
+    //* ---------------------------------------
+
+    let mut state = CachedState::new(state_reader, Some(contract_class_cache));
+
+    //* ------------------------------------
+    //*    Create execution entry point
+    //* ------------------------------------
+
+    let calldata = [1.into()].to_vec();
+    let caller_address = Address(0000.into());
+    let entry_point_type = EntryPointType::External;
+
+    let exec_entry_point = ExecutionEntryPoint::new(
+        address,
+        calldata.clone(),
+        increase_balance_selector.clone(),
+        caller_address,
+        entry_point_type,
+        Some(CallType::Delegate),
+        Some(class_hash),
+    );
+
+    //* --------------------
+    //*   Execute contract
+    //* ---------------------
+    let general_config = StarknetGeneralConfig::default();
+    let tx_execution_context = TransactionExecutionContext::new(
+        Address(0.into()),
+        Felt::zero(),
+        Vec::new(),
+        0,
+        10.into(),
+        general_config.invoke_tx_max_n_steps(),
+        TRANSACTION_VERSION,
+    );
+    let mut resources_manager = ExecutionResourcesManager::default();
+    let expected_key = calculate_sn_keccak("balance".as_bytes());
+
+    let mut expected_accessed_storage_keys = HashSet::new();
+    expected_accessed_storage_keys.insert(expected_key);
+
+    let expected_call_info = CallInfo {
+        caller_address: Address(0.into()),
+        call_type: Some(CallType::Delegate),
+        contract_address: Address(1111.into()),
+        entry_point_selector: Some(increase_balance_selector),
+        entry_point_type: Some(EntryPointType::External),
+        calldata,
+        retdata: [].to_vec(),
```

Comment:
why an empty retdata? If it calls to the balance entrypoint it should return something (at least a 0)

Reply:
The test calls increase balance function
```cairo
@external
func increase_balance{
    syscall_ptr: felt*,
    pedersen_ptr: HashBuiltin*,
    range_check_ptr,
}(amount: felt) {
    let (res) = balance.read();
    balance.write(amount);
    return ();
}
```
which does not have a return type. I'm unsure if this is correct but I should not be expecting return a value, right?

Reply:
Correct, the function effectively returns `void` (or the unit type `()` in Rust).